packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts

```diff
@@ -48,13 +48,24 @@ export const useInfoLabel_unstable = (props: InfoLabelProps, ref: React.Ref<HTML
     required: !!info,
     defaultProps: {
       id: useId('infobutton-'),
-      info,
       size,
+      info,
+    },
+  });
+
+  const infoButtonInfo = resolveShorthand(infoButton?.info, {
+    defaultProps: {
+      id: useId('infobutton-info-'),
     },
   });
 
   if (infoButton) {
+    infoButton.info = infoButtonInfo;
     infoButton['aria-labelledby'] ??= `${label.id} ${infoButton.id}`;
+
+    if (infoButton['aria-owns'] === undefined) {
+      root['aria-owns'] = infoButton.info?.id;
+    }
```

Comment:
This needs to check `root`, not `infoButton`:
```suggestion
    if (root['aria-owns'] === undefined) {
      root['aria-owns'] = infoButton.info?.id;
    }
```

Or you could skip the if statement and use `??=` (like with the `aria-labelledby` above):
```suggestion
    root['aria-owns'] ??= infoButton.info?.id;
```